problems/lib/problem3.rb

```diff
@@ -0,0 +1,14 @@
+class Problem3
+  def has_duplicate(array)
+    len = array.size
+
+    for i in (0..len-1)
+      relocated_index = array[i] % len
+      return true if array[relocated_index] > len

```

Comment:
Prefer having 1 return per method -> at the end of the method. Also `return` keyword is rarely -> never used